matlab/JSBSim_SFunction.cpp

```diff
@@ -0,0 +1,620 @@
+ * JSBSimInterface written by Agostino De Marco for use in the JSBSimMexFunction project. Additional functions have been added and changes 
+ * made to work with SFunction API. Thanks to Agostino for providing the basis for this project.
+ * *************************************************************************************************************************
+
+*/
+#ifdef __cplusplus
+ 
+#endif       // defined within this scope
+#define S_FUNCTION_NAME  JSBSim_SFunction
+#define S_FUNCTION_LEVEL 2
+
+/*
+ * Need to include simstruc.h for the definition of the SimStruct and
+ * its associated macro definitions.
+ */
+#include "simstruc.h"
+#include "mex.h"
+#include "mclcppclass.h"
+#include "matrix.h"
+#include <iostream>
+#include <string>
+#include <vector>
+#include <FGFDMExec.h>
+#include <models/FGPropagate.h>
+#include <models/FGAuxiliary.h>
+#include <models/FGFCS.h>
+#include "JSBSimInterface.h"
+
+
+// 12 States of Initial Condition Vector
+#define u_fps			mxGetPr(ssGetSFcnParam(S, 1))[0]
+#define v_fps			mxGetPr(ssGetSFcnParam(S, 1))[1]
+#define w_fps			mxGetPr(ssGetSFcnParam(S, 1))[2]
+
+#define p_radsec          mxGetPr(ssGetSFcnParam(S, 1))[3]
+#define q_radsec          mxGetPr(ssGetSFcnParam(S, 1))[4]
+#define r_radsec          mxGetPr(ssGetSFcnParam(S, 1))[5]
+
+#define h_sl_ft				mxGetPr(ssGetSFcnParam(S, 1))[6]
+#define long_gc_deg			mxGetPr(ssGetSFcnParam(S, 1))[7]
+#define lat_gc_deg			mxGetPr(ssGetSFcnParam(S, 1))[8]
+
+#define phi_rad			mxGetPr(ssGetSFcnParam(S, 1))[9]
+#define theta_rad		mxGetPr(ssGetSFcnParam(S, 1))[10]
+#define psi_rad			mxGetPr(ssGetSFcnParam(S, 1))[11]
+
+
+// Initial conditions for the controls 
+#define throttle         mxGetPr(ssGetSFcnParam(S, 2))[0]
+#define aileron              mxGetPr(ssGetSFcnParam(S, 2))[1]
+#define elevator              mxGetPr(ssGetSFcnParam(S, 2))[2]
+#define rudder              mxGetPr(ssGetSFcnParam(S, 2))[3]
+#define mixture             mxGetPr(ssGetSFcnParam(S, 2))[4]
+#define runset              mxGetPr(ssGetSFcnParam(S, 2))[5]
+#define flaps              mxGetPr(ssGetSFcnParam(S, 2))[6]
+#define gear              mxGetPr(ssGetSFcnParam(S, 2))[7]
+
+#define delta_t				mxGetPr(ssGetSFcnParam(S, 3))[0] // delta T
+#define script_bool				mxGetPr(ssGetSFcnParam(S, 3))[1] //1 if use script 
+//#define verbosity			ssGetSFcnParam(S, 4) //Verbosity parameter
+#define ac_name				ssGetSFcnParam(S, 0) //Name of JSBSim aircraft model file to load
+#define reset_name          ssGetSFcnParam(S, 5) //Reset name 
+//#define multiplier			mxGetPr(ssGetSFcnParam(S, 5))[0] //JSBSim multiplier
+#define script_name         ssGetSFcnParam(S, 4)
+
+
+// Necessary to create mxArray of initial conditions 
+#define NUMBER_OF_STRUCTS (sizeof(ic)/sizeof(struct init_cond))
+#define NUMBER_OF_FIELDS (sizeof(field_names)/sizeof(*field_names))
+
+JSBSim::FGFDMExec FDMExec;	// Instantiate new JSBSim FDMExec object
```

Comment:
```suggestion
```
Don't declare `FDMExec` a global variable. You'd rather make it a member of `JSBSimInterface`